packages/cli/src/cmds/beacon/initBeaconState.ts

```diff
@@ -13,10 +16,59 @@ import {
   initStateFromEth1,
 } from "@chainsafe/lodestar";
 import {getStateTypeFromBytes} from "@chainsafe/lodestar/lib/util/multifork";
+import {Checkpoint} from "@chainsafe/lodestar-types/phase0";
 import {downloadOrLoadFile} from "../../util";
 import {IBeaconArgs} from "./options";
 import {defaultNetwork, IGlobalArgs} from "../../options/globalOptions";
 import {getGenesisFileUrl} from "../../networks";
+import {WeakSubjectivityServers} from "../weakSubjectivityState";
+
+type WSResponse = {
+  current_epoch: number;
+  ws_checkpoint: string;
+  ws_period: number;
+  is_safe: boolean;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ws_state: any;
+};
+
+type WeakSubjectivityData = {
+  state: TreeBacked<allForks.BeaconState>;
+  checkpoint: Checkpoint;
+};
+
+async function getWeakSubjectivityData(
+  config: IBeaconConfig,
+  args: IBeaconArgs & IGlobalArgs,
+  server: string,
+  logger: ILogger
+): Promise<WeakSubjectivityData> {
+  logger.info("Fetching weak subjectivity state from ChainSafe at " + server);
+  const response = await got(server, {searchParams: {checkpoint: args.weakSubjectivityCheckpoint}});
+  const responseBody = JSON.parse(response.body) as WSResponse;
+  const data = responseBody.ws_state.data;
```

Comment:
can ws_state be null? If yess you should add null checkhere

Reply:
This ultimately comes from calling the `debug/states/<stateId>` endpoint, so if the stateId is not resolved, the API will throw an error when we try to fetch it.